environments/gcp/README.md

```diff
@@ -0,0 +1,99 @@
+2. Run ``gcloud auth application-default login``
+3. Confirm the ID of your project:
+    ```bash
+    gcloud projects list
+    ```
+
+3 . Set the default project
+
+```bash
+gcloud config set project <YOUR_PROJECT_ID>
+```
+
+4. Set the default region
+    - ``gcloud config set compute/region <your-GCP-region> ``
+    - [Learn more](https://cloud.google.com/compute/docs/gcloud-compute)
+5.  Enable necessary APIs:
+    - ``gcloud services enable servicenetworking.googleapis.com container.googleapis.com compute.googleapis.com``
+6. Create a bucket in the project and in the same region, leaving public access off. Then, go to `terraform.tf`and change the `bucket` for the backend.
+7. Initialize your Terraform environment:
+```bash
+terraform init
+```
+8. Then:
+
+```bash
+terraform plan
+```
+9. Verify changes to be applied and run:
+```bash
+terraform apply
+```
+Example output:
+```bash
+Outputs:
+
+another-pwd = <sensitive>
+db-host = tolist([
+  {
+    "ip_address" = "10.191.0.3"
+    "time_to_retire" = ""
+    "type" = "PRIVATE"
+  },
+])
+db-password = <sensitive>
+gcp-binary-service-account = "flyte-onterraform-flyte-binary"
+gcp-worker-service-account = "flyte-onterraform-flyte-worker"
+gcs_bucket_name = "flyte-onterraform-data"
+gke-cluster-name = "flyte-onterraform"
+```
+
+**Once everything is installed**:
+
+1. Generate the `kubeconfig` entry for your new GKE cluster:
+
+```bash
+gcloud container clusters get-credentials <gke-cluster-name> --region <your-GCP-region> --project <your-project_id>
+```
+
+2. Obtain your PostgreSQL database password:
+```bash
+terraform output db-password
+```
+3. Update the `values-gcp-binary.yaml` file with the Terraform outputs and GCP project info. Especifically, these sections need to be updated:
+
+| Key      | Value |
+| ----------- | ----------- |
+| userSettings.dbHost      | `db-host` Terraform output       |
+|userSettings.googleProjectID | Your GCP project ID|
+| userSettings.hostname   | The FQDN you will use to connect to Flyte        |
+|userSettings.IAMWorkerRole |`gcp-worker-service-account` Terraform output |
+|userSettings.IAMBackendRole|`gcp-binary-service-account` Terraform output|
+|configuration.database.password |`db-password` Terraform output |
+|configuration.storage.metadataContainer |`gcs_bucket_name` Terraform output |
+|configuration.storage.userDataContainer | `gcs_bucket_name` Terraform output|
+|configuration.storage.providerConfig.project|Your GCP project ID*|
+
+> NOTE: a current limitation in the `flyte-binary` chart, creates this need to use the GCP project ID in two different ways in the values file.
+
+4. Save your changes and run:
+```bash
+helm install flyte-binary flyteorg/flyte-binary --values values-gcp-binary.yaml 
+```
+5. Update the DNS A entry to point to the Ingress IP:
+    - `kubectl get ingress -n flyte`
+6. Update your `$HOME\.flyte\config,yaml` and make `endpoint` your DNS name:
+```yaml
+...
+#Example
+endpoint: dns://flyte-on-gcp.uniondemo.run 
+```
```

Comment:
Instead of having the user run `terraform apply`, copy the outputs into the `values-gcp-binary.yaml`, and then run `helm install`, you could do the following:

```


resource "helm_release" "flyte_binary" {
  values = [
    "${file(".../values-gcp.-binary.yaml")}"
  ]
  
  # Automatically inject the values the user currently is asked to copy manually:
  set {
    name  = "userSettings.dbHost"
    value = module.flyte-db.instance_ip_address
  }

  set {
}
```

This way, everything might work with a single `terraform apply`.

(The only issue can arise when a `helm_release` requires the presence of CRDs in the cluster for terraform to plan. This causes a chicken egg problem but this shouldn't be the case for Flyte.)

Reply:
Right! I'm working right now with `yamlencode` and `templatefile` to achieve exactly this. 

The issue, though, is that `flyte-core` is already prepared for this level of interpolation but for `flyte-binary` there are sections (like `storage`) that don't, so to achieve this for both charts, changes will have to be done there. I'll capture this on an Issue as it's ongoing work that will be ready in another PR

Reply:
Completed